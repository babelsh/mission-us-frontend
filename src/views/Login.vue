<template>
  <v-container text-xs-center>
    <v-layout align-center justify-center>
      <v-flex xs12>
        <p class="headline">Bem-vindo ao Missão Nós!</p>
      </v-flex>
    </v-layout>
    <v-layout align-center justify-center>
      <v-flex>
        <img src="../images/logo.png" height="150px">
      </v-flex>
    </v-layout>
    <v-layout align-center justify-center>
      <v-flex md6 lg6 xs10>
        <v-form>
          <v-text-field label="E-mail" required autofocus />
          <v-text-field
            label="Senha"
            required
            persistent-hint
            :type="showPassword ? 'text' : 'password'"
            :append-icon="showPassword ? 'visibility_off' : 'visibility'"
            @click:append="showPassword = !showPassword"
          />
        </v-form>
      </v-flex>
    </v-layout>
    <v-layout align-center justify-center>
      <v-flex xs10 lg2>
        <v-btn color="primary" block>Prosseguir</v-btn>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
export default {
  name: 'Login',
  data() {
    return {
      showPassword: false,
    };
  },
}
</script>

<style scoped>

</style>
