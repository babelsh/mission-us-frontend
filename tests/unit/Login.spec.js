import { shallowMount } from '@vue/test-utils';
import Login from '@/views/Login.vue';
import Vue from 'vue';
import Vuetify from 'vuetify';

describe('Login', () => {
  let wrapper;

  beforeEach(() => {
    Vue.use(Vuetify);
  });

  test('Login is a Vue instance', () => {
    wrapper = shallowMount(Login);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('renders the component properly', () => {
    wrapper = shallowMount(Login);

    expect(wrapper).toMatchSnapshot();
  });
});
