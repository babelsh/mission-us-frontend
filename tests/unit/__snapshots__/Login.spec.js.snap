// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders the component properly 1`] = `
<vcontainer-stub tag="div" text-xs-center="">
  <vlayout-stub tag="div" align-center="" justify-center="">
    <vflex-stub tag="div" xs12="">
      <p class="headline">Bem-vindo ao Missão Nós!</p>
    </vflex-stub>
  </vlayout-stub>
  <vlayout-stub tag="div" align-center="" justify-center="">
    <vflex-stub tag="div"><img src="../images/logo.png" height="150px"></vflex-stub>
  </vlayout-stub>
  <vlayout-stub tag="div" align-center="" justify-center="">
    <vflex-stub tag="div" md6="" lg6="" xs10="">
      <vform-stub>
        <vtextfield-stub label="E-mail" required="" autofocus=""></vtextfield-stub>
        <vtextfield-stub label="Senha" required="" persistent-hint="" type="password" append-icon="visibility"></vtextfield-stub>
      </vform-stub>
    </vflex-stub>
  </vlayout-stub>
  <vlayout-stub tag="div" align-center="" justify-center="">
    <vflex-stub tag="div" xs10="" lg2="">
      <vbtn-stub activeclass="v-btn--active" block="true" ripple="true" tag="button" type="button" color="primary">Prosseguir</vbtn-stub>
    </vflex-stub>
  </vlayout-stub>
</vcontainer-stub>
`;
